api4/system.go

```diff
@@ -339,7 +339,7 @@ func postLog(c *Context, w http.ResponseWriter, r *http.Request) {
 	forceToDebug := false
 
 	if !*c.App.Config().ServiceSettings.EnableDeveloper {
-		if c.AppContext.Session().UserId == "" {
+		if c.AppContext.Session().UserId == "" || c.AppContext.Scopes() != nil {
```

Comment:
Can you please elaborate on this change, @larkox ?

Reply:
`nil` in this context means "it is a legacy app, or no app". If the session is not coming from an app or the app is legacy, act as a normal user.

If not, go safe, and act as if you had no permission for this endpoint.